lastgersync/processor.go

```diff
@@ -106,22 +85,57 @@ func (p *processor) ProcessBlock(ctx context.Context, block sync.Block) error {
 	if err := tx.Commit(); err != nil {
 		return err
 	}
-	shouldRollback = false
 	p.log.Debugf("processed %d events until block %d", len(block.Events), block.Num)
 	return nil
 }
 
+// GetLastProcessedBlock retrieves the most recent block processed by the processor,
+// including those without events.
+func (p *processor) GetLastProcessedBlock(ctx context.Context) (uint64, error) {
+	var block BlockNum
+	if err := meddler.QueryRow(
+		p.db,
+		&block,
+		"SELECT num FROM block ORDER BY num DESC LIMIT 1;",
+	); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return 0, nil
+		}
+		return 0, err
+	}
+	return block.Num, nil
+}
+
+// GetLastIndex retrieves the highest L1InfoTreeIndex recorded in the imported_global_exit_root table
+func (p *processor) getLastIndex() (uint32, error) {
+	var lastIndex uint32
+	row := p.db.QueryRow(`
+		SELECT l1_info_tree_index 
+		FROM imported_global_exit_root 
+		ORDER BY l1_info_tree_index DESC LIMIT 1;
+	`)
+	err := row.Scan(&lastIndex)
```

Comment:
Optional nitpick, move the declaration closer to its usage.

```suggestion
	row := p.db.QueryRow(`
		SELECT l1_info_tree_index 
		FROM imported_global_exit_root 
		ORDER BY l1_info_tree_index DESC LIMIT 1;
	`)
    var lastIndex uint32
	err := row.Scan(&lastIndex)
```